### PR TITLE
[4.0] Add prepared statements to plugin privacy actionlogs

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the plugin in all ways you can think of.

* Create users
* Generate actionlog entries
* Try to export the action log.

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/privacy/actionlogs/actionlogs.php [hunk 1]
```diff
@@ -7,6 +7,8 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\Database\ParameterType;
```
Comment (Quy) [inline plugins/privacy/actionlogs/actionlogs.php]: Move after `defined`.

File: plugins/privacy/actionlogs/actionlogs.php [hunk 2]
```diff
@@ -36,12 +37,14 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
...
 		$domain = $this->createDomain('user_action_logs', 'joomla_user_action_logs_data');
+		$db     = $this->db;
 
-		$query = $this->db->getQuery(true)
-			->select('a.*, u.name')
-			->from('#__action_logs AS a')
-			->innerJoin('#__users AS u ON a.user_id = u.id')
-			->where($this->db->quoteName('a.user_id') . ' = ' . $user->id);
+		$query = $db->getQuery(true)
+			->select($db->quoteName(['a.*', 'u.name']))
```
Comment (SharkyKZ) [inline plugins/privacy/actionlogs/actionlogs.php]: Can't use `quoteName()` on `a.*`.

File: plugins/privacy/actionlogs/actionlogs.php [hunk 3]
```diff
@@ -36,12 +37,14 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
...
+			->from($db->quoteName('#__action_logs', 'a'))
+			->innerJoin($db->quoteName('#__users', 'u'), $db->quoteName('a.user_id') . ' = ' . $db->quoteName('u.id'))
+			->where($this->db->quoteName('a.user_id') . ' = :id')
```
Comment (SharkyKZ) [inline plugins/privacy/actionlogs/actionlogs.php]: For consistency here.
```suggestion
			->where($db->quoteName('a.user_id') . ' = :id')
```

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on f2d0ebd4436ea10c37c7f35c4fa5b018fe332b9b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25409">issues.joomla.org/tracker/joomla-cms/25409</a>.</sub>

File: plugins/privacy/actionlogs/actionlogs.php [hunk 4]
```diff
@@ -36,16 +37,18 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
...
 
-		$query = $this->db->getQuery(true)
-			->select('a.*, u.name')
-			->from('#__action_logs AS a')
-			->innerJoin('#__users AS u ON a.user_id = u.id')
-			->where($this->db->quoteName('a.user_id') . ' = ' . $user->id);
+		$query = $db->getQuery(true)
+			->select(['a.*', $db->quoteName('u.name')])
+			->from($db->quoteName('#__action_logs', 'a'))
+			->innerJoin($db->quoteName('#__users', 'u'), $db->quoteName('a.user_id') . ' = ' . $db->quoteName('u.id'))
```
Comment (Quy) [inline plugins/privacy/actionlogs/actionlogs.php]: Change innerJoin to join.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 1db70190b3574594fcfcbe35192fe24569343bd6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25409">issues.joomla.org/tracker/joomla-cms/25409</a>.</sub>

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 1db70190b3574594fcfcbe35192fe24569343bd6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25409">issues.joomla.org/tracker/joomla-cms/25409</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25409">issues.joomla.org/tracker/joomla-cms/25409</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**